app/ui/src/main/java/com/fsck/k9/activity/setup/AccountSetupBasics.java

```diff
@@ -263,6 +266,34 @@ private void finishAutoSetup(ConnectionSettings connectionSettings) {
         AccountSetupCheckSettings.actionCheckSettings(this, mAccount, CheckDirection.INCOMING);
     }
 
+    private ConnectionSettings providersXmlDiscoveryDiscover(String email, DiscoveryTarget discoveryTarget) {
```

Comment:
I don't really want this to exist for long in it's current form because it is making assumptions about the implementation of ProvidersXmlDiscovery.discover() but it's a stepping stone.

Reply:
I don't think we should try to modify the current setup UI to work with the plan proposed in #4721. It's better to create a new Activity. 
Since I expect it to be quite a while before the new setup experience is complete, I think it'd be good to support both UIs at the same time. We'll use the old UI for releases until the new one is ready. I can set this up next week.
But that also means this code will be around for a while. We should make sure it works like it did before, as best as we can.